pylibftdi/device.py

```diff
@@ -543,24 +546,28 @@ def readline(self, size=0):
         This only works for mode='t' on Python3
         """
         lsl = len(os.linesep)
-        line_buffer = []
+        line_buffer: list[str] = []
         while True:
             next_char = self.read(1)
             if next_char == '' or (0 < size < len(line_buffer)):
                 break
+            if isinstance(next_char, bytes):
+                next_char = self.decoder.decode(next_char)
             line_buffer.append(next_char)
             if (len(line_buffer) >= lsl and
                     line_buffer[-lsl:] == list(os.linesep)):
                 break
         return ''.join(line_buffer)
 
-    def readlines(self, sizehint=None):
+    def readlines(self, sizehint: Optional[int]=None) -> list[str]:
         """
         readlines() for file-like compatibility.
         """
-        lines = []
+        lines: list[str] = []
         if sizehint is not None:
             string_blob = self.read(sizehint)
+            if isinstance(string_blob, bytes):
+                string_blob = self.decoder.decode(string_blob)
```

Comment:
See comments on lines 554-555 above - I think this is analogous and probably better to assume (e.g. with assert / type check and raise exception etc) that `read()` is returning `str`.

Reply:
Added the type check/assertion.